src/main/java/org/prebid/server/bidder/ownadx/OwnAdxBidder.java

```diff
@@ -0,0 +1,141 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class OwnAdxBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpOwnAdx>> OWN_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String X_OPEN_RTB_VERSION = "2.5";
+    private static final String ACCOUNT_ID_MACROS_ENDPOINT = "{{AccountID}}";
+    private static final String ZONE_ID_MACROS_ENDPOINT = "{{ZoneID}}";
+    private static final String SOURCE_ID_MACROS_ENDPOINT = "{{SourceId}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public OwnAdxBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                final ExtImpOwnAdx impOwnAdx = parseImpExt(imp);
+                httpRequests.add(createHttpRequest(bidRequest, impOwnAdx));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpOwnAdx parseImpExt(Imp imp) {
+        final ExtImpOwnAdx extImpOwnAdx;
+        try {
+            extImpOwnAdx = mapper.mapper().convertValue(imp.getExt(), OWN_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Missing bidder ext in impression with id: " + imp.getId());
+        }
+        return extImpOwnAdx;
+    }
+
+    private HttpRequest<BidRequest> createHttpRequest(BidRequest bidRequest, ExtImpOwnAdx extImpOwnAdx) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(updateUrl(extImpOwnAdx))
+                .headers(updateHeader())
+                .body(mapper.encodeToBytes(bidRequest))
+                .impIds(BidderUtil.impIds(bidRequest))
+                .payload(bidRequest)
+                .build();
+    }
+
+    private String updateUrl(ExtImpOwnAdx extImpOwnAdx) {
+        return endpointUrl
+                .replace(ACCOUNT_ID_MACROS_ENDPOINT, extImpOwnAdx.getSspId() != null ? extImpOwnAdx.getSspId() : "")
+                .replace(ZONE_ID_MACROS_ENDPOINT, extImpOwnAdx.getSeatId() != null ? extImpOwnAdx.getSeatId() : "")
+                .replace(SOURCE_ID_MACROS_ENDPOINT, extImpOwnAdx.getTokenId() != null ? extImpOwnAdx.getTokenId() : "");
+    }
+
+    private static MultiMap updateHeader() {
```

Comment:
the `update` word doesn't make sense since you don't pass any parameters to be updated, so probably `makeHeaders` fits better